recipes-security/refpolicy/refpolicy-mcs-2.%/policy/modules/system/stubdom-helpers.te

```diff
@@ -119,3 +126,48 @@ allow audio_helper_t self:shm create_shm_perms;
 init_dontaudit_use_fds(audio_helper_t)
 init_dontaudit_use_script_fds(audio_helper_t)
 xc_dontaudit_read_etc_files(audio_helper_t)
+
+########################################
+#
+# Local Policy: varstored
+#
+gen_require(`
+	type xen_device_t, xenstored_t, xenstored_var_run_t;
+')
+
+# varstored-watch executing & monitoring varstored
+allow varstored_t bin_t:dir search;
+allow varstored_t initrc_t:fd use;
```

Comment:
This should probably be a dontaudit.

Reply:
Can you elaborate on this one? Is this denial typically not blocking?

Reply:
tl:dr: I think you want to add `init_dontaudit_use_script_fds(varstored_t)`

I think an FD from the initscript leaks into xenmgr which then leaks to xl and to varstored/watch.  Maybe varstored/watch's use of stdio is part of the reason?  Anyway, we know varstored/watch ware self contained and don't rely on using inherited FDs, so we can just dontaudit them.  That way they are disallowed but don't log avc denials.

Cleaning up xenmgr leaking FDs to xl and xl leaking FDs to helpers is another option... but dontaudit is fine for now.